### ISSUE TITLE
[Feature request] Add a menu option for the shortest path in km instead of the fastest.

### ISSUE THREAD

**Issue by ytrezq** [osmandapp/Osmand#7584]
All Groups from Pedestrian to Public Transport could use such option.

This would be especially meaningful for Train or Car where some national train companies bill their tickets according to distance and some car assurances also bill according to distance. And sometime a Toll motorway shorten the distance on a part of the track.

I tried the routing xml file but I’m still getting paths following the fastest path and pedestrian mode paths rejecting motorways and taking track routes.

**Comment by Jorgeno** [comment]
I see that the estimated travelling time is much longer when the algorithm presents an identical route also with fuel economy enabled. But the time presented without fuel economy enabled seems reasonable.
Is this really correct, suggesting that one should drive much much slower in that case or is it a bug (perhaps confusing optimization variables in the algorithm with the actual travelling time)?

The result is that when the fuel optimized route is not identical to the fastest it becomes really difficult to evaluate it as the presented travelling time much too long, i.e. the trade-off against the fastest route becomes irrelevant.

**Comment by ytrezq** [comment]
@Jorgeno : I’m not talking about the fastest path, but the shortest length. Sometimes the fuel economy route isn’t showing the shortest length because there are cases where you burn more fuel at doing so.

What I want the most is the shortest path for cars even if it burns more fuel because when I’m renting a car, I’m billed according to the number of Km used regardless of the speed.

**Comment by Jorgeno** [comment]
Yes, I see what you mean. Sorry for confusing the topic off your thread.
My concern is instead better explained by this post: https://github.com/osmandapp/Osmand/issues/9345
It seems to be an actual bug and not a missing feature.

**Comment by ytrezq** [comment]
@Jorgeno : it’s unrelated. I don’t care about the time travel. What I’m seeing is the shortest path indeed cost way more fuel on the route I’m taking so there’s nothing wrong with fuel economy mode adding 90Km more to the shortest route.

It’s a different problem. This option for car simply doesn’t exist yet. Though having it for bikes is an improvement since the time that issue was created.

**Comment by vshcherb** [comment]
The issue that shortest path for vehicle is calculated using a constant speed, so fastest route = shortest. For pedestrian and mostly bicycle the same speed is specified so by definition fastest = shortest route.

In order to make the difference different roads should have different speed and today it is only possible by creating custom pedestrian profile

**Comment by ytrezq** [comment]
@vshcherb : there’s more than that. Additionally, there are road type preferences. So typically, a pedestrian path will favor tertiary and trunk roads first over primary one even if it makes the route longer.

The most striking examples is this route using bike mode of ᴏꜱᴍᴀɴᴅ+ in terms shortest path first of driving style and elevation data disabled (please pay attention to the details at the left) :
![](https://user-images.githubusercontent.com/3824869/86308554-e925e300-bc19-11ea-96a8-f6f895079fa6.png)

Which can be compared to the car mode of this provider (because even Google ᴍᴀᴘꜱ is producing longer routes though) :
![Screenshot_20200701-203153](https://user-images.githubusercontent.com/3824869/86308545-e0cda800-bc19-11ea-9354-25c5aa3e5459.jpg)
With the last one still not ideal because it bans Motorways whereas the first part (since in the remaining there’s not even a primary route) shows there’s a line being drawn whereas the non motorway route for the first part takes lot of turns (and thus is likely to make the path longer).

**Comment by miklcct** [comment]
It's not strictly true for pedestrians where shortest = fastest because, if the shortest route involves a climb, it may be faster to go around the hill.

**Comment by Noitarud** [comment]
Yeah, you could be driving a crane or tractor…
